File: 2022/day6.py
from more_itertools import sliding_window
from itertools import takewhile
from functools import reduce
from operator import add

print(
    reduce(
        add,
        map(
            bool,
            takewhile(
                lambda packet: len(set(packet)) != len(packet),
                sliding_window(open("./input6.txt").read(), w:=4),
            ),
        ),
        w)
)

File: 2022/day6.rs
use itertools::Itertools;

fn main() {
    let w = 14;
    let lines = include_str!("../input6.txt")
        .chars()
        .collect::<Vec<_>>()
        .windows(w)
        .take_while(|x| !x.iter().all_unique())
        .count() + w;
    println!("{:#?}", lines)
}
